tony-history-server/app/controllers/HomeController.java

```diff
@@ -0,0 +1,102 @@
+package controllers;
+
+import com.typesafe.config.Config;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import javax.inject.Inject;
+
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hdfs.HdfsConfiguration;
+import org.apache.hadoop.security.UserGroupInformation;
+import org.apache.hadoop.yarn.api.ApplicationConstants;
+import org.apache.hadoop.yarn.conf.YarnConfiguration;
+
+import play.Logger.ALogger;
+import play.Logger;
+import play.mvc.*;
+
+
+public class HomeController extends Controller {
+  private static final ALogger LOG = Logger.of(HomeController.class);
+  private final Config config;
+
+  private final String HADOOP_CONF_DIR = ApplicationConstants.Environment.HADOOP_CONF_DIR.key();
+  private final String CORE_SITE_CONF = YarnConfiguration.CORE_SITE_CONFIGURATION_FILE;
+  private final String HDFS_SITE_CONF = "hdfs-site.xml";
+
+  @Inject
+  public HomeController(Config config) {
+    this.config = config;
+  }
+
+  private HdfsConfiguration setUpHdfsConf() {
+    HdfsConfiguration conf = new HdfsConfiguration();
+
+    if (System.getenv("HADOOP_CONF_DIR") != null) {
+      conf.addResource(new Path(System.getenv(HADOOP_CONF_DIR) + File.separatorChar + CORE_SITE_CONF));
+      conf.addResource(new Path(System.getenv(HADOOP_CONF_DIR) + File.separatorChar + HDFS_SITE_CONF));
+    }
+
+    // return `kerberos` if on Kerberized cluster.
+    // return `simple` if develop locally.
+    LOG.debug("Hadoop Auth Setting: " + conf.get("hadoop.security.authentication"));
+    return conf;
+  }
+
+  private void setUpKeytab(HdfsConfiguration hdfsConf) {
+    boolean isSecurityEnabled = hdfsConf.get("hadoop.security.authentication").equals("kerberos");
+    if (isSecurityEnabled) {
+      try {
+        UserGroupInformation.setConfiguration(hdfsConf);
+        UserGroupInformation.loginUserFromKeytab(config.getString("keytab.user"), config.getString("keytab.location"));
+      } catch (IOException e) {
+        LOG.error("Failed to set up keytab", e);
+      }
+    }
+  }
+
+  private FileSystem getFileSystem(HdfsConfiguration hdfsConf) {
```

Comment:
normally these functions which has nothing to do with current view should be moved to a Utils class as Controller class are the first ones to explode in size.

Reply:
Agreed. We can probably create a HadoopUtils class and move all these utility functions there.

Reply:
We can probably do that when the app gets more complex with more views/controllers. Right now, I think it's ok to include these in HomeController.

Reply:
It should not be included at HomeController.. these would be needed across different controllers next month. I'd suggest just do that in this PR.